react-native-mcu-manager/src/Upgrade.ts

```diff
@@ -91,16 +97,24 @@ class Upgrade {
    */
   constructor(
     bleId: string,
-    updateFileUriString: string,
-    updateOptions: UpgradeOptions
+    updateFileUri: string,
+    updateOptions: UpgradeOptions,
+    private onProgress?: (progress: number) => void,
+    private onStateChange?: (state: string) => void
   ) {
-    this.id = uuidv4() as UpgradeID;
+    this.id = generateUUID() as UpgradeID;
```

Comment:
Not really any need for this to be in UUID form...

We could switch this out for a "large enough" int?

```suggestion
    this.id = Math.floor(100000000 + Math.random() * 900000000);
```

Reply:
Do we still need it to be a string?